src/pages/ca-adjustment/form/CAadjustmentForms.js

```diff
@@ -0,0 +1,319 @@
+import CustomTextArea from 'src/components/Inputs/CustomTextArea'
+import { useDocumentType } from 'src/hooks/documentReferenceBehaviors'
+
+export default function CAadjustmentForm({ labels, access, recordId, functionId }) {
+  const [editMode, setEditMode] = useState(!!recordId)
+
+  const { documentType, maxAccess, changeDT } = useDocumentType({
+    functionId: functionId,
+    access: access,
+    enabled: !recordId
+  })
+
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const invalidate = useInvalidate({
+    endpointId: CashBankRepository.CAadjustment.page
+  })
+
+  const { formik } = useForm({
+    initialValues: {
+      recordId: recordId || null,
+      reference: '',
+      name: '',
+      dtId: documentType?.dtId,
+      plantId: '',
+      date: new Date(),
+      currencyId: '',
+      status: '',
+      cashAccountId: '',
+      amount: '',
+      baseAmount: '',
+      exRate: '',
+      rateCalcMethod: '',
+      functionId: functionId,
+      notes: ''
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      amount: yup.string().required(' '),
+      currencyId: yup.string().required(' '),
+      cashAccountId: yup.string().required(' '),
+      date: yup.string().required(' ')
+    }),
+    onSubmit: async obj => {
+      const recordId = obj.recordId
+      if (!recordId) {
+        obj.baseAmount = obj.amount
+        obj.status = 1
+        obj.rateCalcMethod = 1
+        obj.exRate = 1
+      }
+
+      const response = await postRequest({
+        extension: CashBankRepository.CAadjustment.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!recordId) {
+        toast.success('Record Added Successfully')
+        formik.setValues({
+          ...obj,
+          baseAmount: obj.amount,
+
+          recordId: response.recordId
+        })
+      } else {
+        toast.success('Record Edited Successfully')
+      }
+
+      setEditMode(true)
+      try {
+        const res = await getRequest({
+          extension: CashBankRepository.CAadjustment.get,
+          parameters: `_recordId=${response.recordId}`
+        })
+
+        formik.setFieldValue('reference', res.record.reference)
+      } catch (error) {}
+      invalidate()
+    }
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        if (recordId) {
+          const res = await getRequest({
+            extension: CashBankRepository.CAadjustment.get,
+            parameters: `_recordId=${recordId}`
+          })
+
+          formik.setValues({
+            ...res.record,
+
+            date: formatDateFromApi(res.record.date)
+          })
+        }
+      } catch (exception) {}
+    })()
+  }, [])
+
+  const onPost = async () => {
+    try {
+      const res = await postRequest({
+        extension: CashBankRepository.CAadjustment.post,
+        record: JSON.stringify(formik.values)
```

Comment:
This will be submitted regardless of validation. Make sure this is what you want.